pkg/engine/context/deferred.go

```diff
@@ -34,7 +37,11 @@ func (dl *deferredLoader) HasLoaded() bool {
 }
 
 func (dl *deferredLoader) LoadData() error {
-	return dl.loader.LoadData()
+	if err := dl.loader.LoadData(); err != nil {
+		dl.logger.Error(err, "Failed to load data: ")
```

Comment:
Please start log messages with a lower case. It would also be helpful to add the name.

```
dl.logger.Errorf(err, "failed to load data for %s", dl.name)
```